### DISCUSSION TITLE
Transferring the binary file from the local system to the docker container

### DISCUSSION THREAD

**Discussion by VahidYm** [prisma/prisma-client-go#1215] [General]
I'm currently working with Prisma Client Go and I have a question regarding the generation of the Prisma binary file (prisma-query-engine-debian-openssl-1.1.x). 

Is it feasible to generate this binary file on my local system (which is running Ubuntu 20.04) and then transfer it to a different server? The motivation behind this is to minimize the download and generation time of the binary file within a Docker container. 

Any guidance or suggestions would be greatly appreciated. Thank you.

**Comment by steebchen** [comment]
It's definitely not recommended to add something locally generated to the docker container to ensure reproducible images. So you would just have to accept the download times. You can check [the docker example](https://goprisma.org/docs/reference/deploy/docker) which uses the `prefetch` command which caches the binary generation step, so that if you just have application code changes it would use the cache.

If you really want to generate the binaries locally and then copy them to your dockerfile, you can do so by adding `binaryTargets = ["native", "debian-openssl-1.1.x"]` to your generator config in your prisma.schema file. To find out which exact value to use for the target, you can check the [Prisma binaryTargets docs](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#binarytargets-options).

**Reply by VahidYm** [reply]
Thank you for your prompt and informative reply 🙏
